File: frontend/src/types/fieldMapping.ts
export type FieldKey = "attendee_name" | "workshop_title" | "certificate_date";

export interface FieldMapping {
  id?: number;
  template_id?: number;
  field_key: FieldKey;
  x: number;
  y: number;
  width: number;
  height: number;
  font_family: string | null;
  font_size: number | null;
  font_color: string | null;
  font_weight: string | null;
  alignment: "left" | "center" | "right" | null;
  visible: boolean;
}

File: backend/app/schemas/field_mapping.py
from pydantic import BaseModel, ConfigDict, Field


class FieldMappingBase(BaseModel):
    field_key: str
    x: float = Field(ge=0)
    y: float = Field(ge=0)
    width: float = Field(gt=0)
    height: float = Field(gt=0)
    font_family: str | None = "helv"
    font_size: float | None = Field(default=36, gt=0)
    font_color: str | None = "#111827"
    font_weight: str | None = "normal"
    alignment: str | None = "center"
    visible: bool = True


class FieldMappingCreate(FieldMappingBase):
    pass


class FieldMappingUpdate(BaseModel):
    mappings: list[FieldMappingCreate]


class FieldMappingRead(FieldMappingBase):
    id: int
    template_id: int

    model_config = ConfigDict(from_attributes=True)
